### PR TITLE
chore: add ruin_me powershell tests

### PR DESCRIPTION
**Pull request by Gankra**

### CODE REVIEW

File: cargo-dist/tests/gallery/dist.rs [hunk 1]
```diff
@@ -366,6 +369,221 @@ impl DistResult {
...
+                assert_eq!(
+                    empirical_path.lines().next().unwrap(),
+                    bin_path.as_str(),
+                    "bin path wasn't right"
+                );
+            }
+
+            // check the install receipts
+            // FIXME: temporarily disabled for the feature being broken
+            // self.check_install_receipt(ctx, &bin_dir, &receipt_file, ".exe");
```
Comment (Gankra) [inline cargo-dist/tests/gallery/dist.rs:582]: todo for @mistydemeo: fix install receipts and uncomment this line

Comment (mistydemeo) [reply cargo-dist/tests/gallery/dist.rs:608]: Rebase with this removed works: https://github.com/axodotdev/cargo-dist/pull/866

Comment (mistydemeo) [reply cargo-dist/tests/gallery/dist.rs:608]: You can reenable this now that #863 is merged.

File: cargo-dist/tests/gallery/dist.rs [hunk 2]
```diff
@@ -366,6 +369,248 @@ impl DistResult {
...
+    }
+
+    // Runs the installer script in a temp dir, attempting to set env vars to contain it to that dir
+    #[allow(unused_variables)]
+    pub fn runtest_powershell_installer(
+        &self,
+        ctx: &TestContext<Tools>,
+        expected_bin_dir: &str,
+    ) -> Result<()> {
+        // Only do this on unix, and only do it if RUIN_MY_COMPUTER_WITH_INSTALLERS is set
```
Comment (mistydemeo) [inline cargo-dist/tests/gallery/dist.rs:439]:
```suggestion
        // Only do this on windows, and only do it if RUIN_MY_COMPUTER_WITH_INSTALLERS is set
```

File: cargo-dist/tests/gallery/dist.rs [hunk 3]
```diff
@@ -366,6 +369,248 @@ impl DistResult {
...
+            }
+            std::fs::create_dir_all(&appdata).unwrap();
+
+            // save the current PATH in the registry
+            let saved_path = run_ps1_script(
+                &powershell,
+                &tempdir,
+                "savepath.ps1",
+                r#"
+            $Item = Get-Item -Path "HKCU:\Environment"
```
Comment (mistydemeo) [inline cargo-dist/tests/gallery/dist.rs:490]: Oh man

File: cargo-dist/tests/gallery/dist.rs [hunk 4]
```diff
@@ -366,6 +369,248 @@ impl DistResult {
...
+                assert!(bin_path.exists(), "{bin_name} wasn't created");
+
+                let bin =
+                    CommandInfo::new(bin_name, Some(bin_path.as_str())).expect("failed to run bin");
+                assert!(bin.version().is_some(), "failed to get app version");
+
+                // checking path...
+                // Make a test.ps1 script that runs `where.exe {bin_name}`
+                //
+                // (note that "where" and "where.exe" are completely different things...)
```
Comment (mistydemeo) [inline cargo-dist/tests/gallery/dist.rs:579]: Oh NO

**Review by mistydemeo — APPROVED** [review]

File: cargo-dist/tests/gallery/dist.rs (hunk 4 above)
Comment (Gankra) [reply cargo-dist/tests/gallery/dist.rs:579]: ✨

**Review by mistydemeo — APPROVED** [review]

**Pull request merged**